File: server-rails-single/app/policies/task_policy.rb
# frozen_string_literal: true

# Single-tenant TaskPolicy — permissive on attributes; ownership enforced by
# Scopes::TaskScope (tasks of the current user's projects). See ProjectPolicy.
class TaskPolicy < Rhino::ResourcePolicy
  def permitted_attributes_for_show(_user) = ["*"]
  def hidden_attributes_for_show(_user) = []
  def permitted_attributes_for_create(_user) = ["*"]
  def permitted_attributes_for_update(_user) = ["*"]
end

File: server-nestjs-single/src/policies/TaskPolicy.ts
import { ResourcePolicy } from '@rhino-dev/rhino-nestjs';

/**
 * Task policy — SINGLE-TENANT (role-free). Ownership inherited via TaskScope
 * (parent Project's userId). Action authorization comes from User.permissions.
 */
export class TaskPolicy extends ResourcePolicy {
  override resourceSlug = 'tasks';

  override permittedAttributesForShow(): string[] {
    return ['*'];
  }
  override hiddenAttributesForShow(): string[] {
    return [];
  }
  override permittedAttributesForCreate(): string[] {
    return ['title', 'description', 'status', 'priority', 'estimatedHours', 'dueDate', 'projectId', 'assignedTo'];
  }
  override permittedAttributesForUpdate(): string[] {
    return ['title', 'description', 'status', 'priority', 'estimatedHours', 'dueDate', 'projectId', 'assignedTo'];
  }
}
